### PR TITLE
Use Null Coalesce Operator

### PR DESCRIPTION
**Pull request by carusogabriel**
Following #6874

### CODE REVIEW

File: lib/Doctrine/ORM/UnitOfWork.php [hunk 1]
```diff
@@ -475,11 +475,7 @@ public function & getEntityChangeSet($entity)
         $oid  = spl_object_id($entity);
         $data = [];
 
-        if (!isset($this->entityChangeSets[$oid])) {
-            return $data;
-        }
-
-        return $this->entityChangeSets[$oid];
+        return $this->entityChangeSets[$oid] ?? $data;
```
Comment (Ocramius) [inline lib/Doctrine/ORM/UnitOfWork.php]: This change is incorrect: the method returns a reference, so an expression is not valid when combined with `return`

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: lib/Doctrine/ORM/UnitOfWork.php (hunk 1 above)
Comment (carusogabriel) [reply lib/Doctrine/ORM/UnitOfWork.php]: Should I rollback so?

Comment (Ocramius) [reply lib/Doctrine/ORM/UnitOfWork.php]: Yeap

File: lib/Doctrine/ORM/Repository/DefaultRepositoryFactory.php [hunk 2]
```diff
@@ -28,11 +28,8 @@ public function getRepository(EntityManagerInterface $entityManager, $entityName
     {
         $repositoryHash = $entityManager->getClassMetadata($entityName)->getClassName() . spl_object_id($entityManager);
 
-        if (isset($this->repositoryList[$repositoryHash])) {
-            return $this->repositoryList[$repositoryHash];
-        }
-
-        return $this->repositoryList[$repositoryHash] = $this->createRepository($entityManager, $entityName);
+        return $this->repositoryList[$repositoryHash] ??
+            $this->repositoryList[$repositoryHash] = $this->createRepository($entityManager, $entityName);
```
Comment (Slamdunk) [inline lib/Doctrine/ORM/Repository/DefaultRepositoryFactory.php]: I don't like this change: it's easy to mess with operator precedence this way.

Comment (Ocramius) [reply lib/Doctrine/ORM/Repository/DefaultRepositoryFactory.php]: This is quite standard

Comment (Slamdunk) [reply lib/Doctrine/ORM/Repository/DefaultRepositoryFactory.php]: Btw in this context I would prefer:
```php
if (!isset($this->repositoryList[$repositoryHash])) {
    $this->repositoryList[$repositoryHash] = $this->createRepository($entityManager, $entityName);
}

return $this->repositoryList[$repositoryHash];
```
So Setter and Getter parts are strictly detached

Comment (Ocramius) [reply lib/Doctrine/ORM/Repository/DefaultRepositoryFactory.php]: I moved all my lazy getters to following:

```php
public function getFoo()
{
    return $this->foo
       ?? $this->foo = $this->makeFoo();
}
```

The intent is clearer and the expression is quite simple

Comment (Ocramius) [reply lib/Doctrine/ORM/Repository/DefaultRepositoryFactory.php]: See for example https://github.com/Ocramius/ProxyManager/blob/81d53b2878f1d1c40ad27270e64b51798485dfc5/src/ProxyManager/Configuration.php :-)

Comment (Slamdunk) [reply lib/Doctrine/ORM/Repository/DefaultRepositoryFactory.php]: Ok then.

@carusogabriel may you move the `??` operator to the new line so conjunctions operators are easy to spot?

Comment (carusogabriel) [reply lib/Doctrine/ORM/Repository/DefaultRepositoryFactory.php]: Sure. I’m on city now. I’ll update my PR at Doctrine’s projects this afternoon 😄

**Comment by carusogabriel** [comment]
@Ocramius @Slamdunk Every requested change done :+1:

File: lib/Doctrine/ORM/Mapping/Driver/DatabaseDriver.php [hunk 3]
```diff
@@ -524,11 +524,10 @@ private function getTablePrimaryKeys(Table $table)
      */
     private function getClassNameForTable($tableName)
     {
-        if (isset($this->classNamesForTables[$tableName])) {
-            return $this->namespace . $this->classNamesForTables[$tableName];
-        }
-
-        return $this->namespace . Inflector::classify(strtolower($tableName));
+        return $this->namespace . (
+            $this->classNamesForTables[$tableName] ??
```
Comment (Majkl578) [inline lib/Doctrine/ORM/Mapping/Driver/DatabaseDriver.php]: This ?? should also be at the beginning of next line.

Comment (carusogabriel) [reply lib/Doctrine/ORM/Mapping/Driver/DatabaseDriver.php]: Should I indent?
```php
$this->classNamesForTables[$tableName]
    ?? Inflector::classify(strtolower($tableName))
```

Comment (Majkl578) [reply lib/Doctrine/ORM/Mapping/Driver/DatabaseDriver.php]: OF course, 4 spaces as usual. 👍

**Review by Majkl578 — APPROVED** [review]

**Comment by Majkl578** [comment]
Waiting for resolution of #6964 as this will likely conflict.

**Comment by Majkl578** [comment]
This will need a rebase now, sorry. :/

**Comment by carusogabriel** [comment]
No problem, I’ll rebase and solve the conflicts

**Review by Majkl578 — APPROVED** [review]

**Pull request merged**